PIL/PngImagePlugin.py

```diff
@@ -687,6 +687,9 @@ def _save(im, fp, filename, chunk=putchunk, check=0):
             mode = "%s;%d" % (mode, bits)
 
     # encoder options
+    if (im.info.get('showing', False)):
+        im.encoderinfo['compress_level'] = 1
+        im.info.pop('showing', None)
```

Comment:
This is not an appropriate separation of concerns. The PNG encoder should not care that it's being called from within the show method. 

A better change would be to the internal `Image._dump` command. It should be altered to take `**options` to be passed into the save command
```
    def _dump(self, file=None, format=None, **options):
        import tempfile
        suffix = ''
        if format:
            suffix = '.'+format
        if not file:
            f, file = tempfile.mkstemp(suffix)
            os.close(f)

        self.load()
        if not format or format == "PPM":
            self.im.save_ppm(file, **options)
        else:
            if not file.endswith(format):
                file = file + "." + format
            self.save(file, format, **options)
        return file
```
(and renaming file to something that does not shadow the builtin)